Add plugin details to zip file errors

A248 opened the pull request:
Recently, some users began reporting an annoying classloading bug, where "IllegalStateException: zip file closed" would travel **across** plugin boundaries. Here are some examples:

* https://pastebin.com/cCzr98sa
* https://pastebin.com/dSAfLcYK

This PR does **NOT** solve any bugs itself. It merely makes it easier to determine which plugin was the source of "zip file closed" in a complex plugin environment where bugs happen sporadically.

That such bugs can cross plugin boundaries at all is reason for this PR

Strokkur424 reviewed (changes requested):
This doesn't solve anything at all and also doesn't particularly help with diagnosing the issue. I think you should instead try and figure out what the cause of the zip file closing is and PR a fix for that.

Commit:
Add plugin details to zip file errors

Helps debug classloading across plugin boundaries. Zip file errors
  can be thrown for multiple reasons, and they are capable of
  affecting other plugins' classloading.
